ui/src/components/SampleView/ContextMenu.jsx

```diff
@@ -486,7 +479,7 @@ export default class ContextMenu extends React.Component {
     const menuOptions = this.menuOptions();
     let optionList = [];
 
-    if (this.props.sampleID !== undefined) {
```

Comment:
I've kept this condition for now, but I don't find it ideal. If there's no sample mounted and you right-click on a shape, you get a different context menu that if you right-click on the same shape and there's a sample mounted.

I would prefer for context menu actions that require a sample to be mounted to be conditionally rendered individually or to show an error to the user when used.

Reply:
> I would prefer for context menu actions that require a sample to be mounted to be conditionally rendered individually or to show an error to the user when used.

Yes, the context menu on a shape without a sample mounted should give an error, just like we can't create a point without a smple

Reply:
Sorry, I've edited my comment to clarify my thought: I think that actions that don't require a sample to be mounted, like _Delete Point_ should be shown when right-clicking on a line regardless of the presence of a sample.

That's because when you mount a sample, perform 3-click centring, save the created point, and unmount the sample, the point remains on the sample view, so you can still right-click on it. At the moment, when you're in this situation, you basically can't delete anything:

![image](https://github.com/user-attachments/assets/88d46ea5-97d6-497b-b685-c4b2ab72a61e)

Reply:
Hm, yes makes sense from that "point" (no pun intended :) ) of view. But all the shapes should actually be cleared when a sample is unmounted. We should look at that if its not the case